mne/beamformer/tests/test_rap_music.py

```diff
@@ -0,0 +1,175 @@
+                                                random_state=rng)
+
+    return sim_evoked, stc
+
+
+@slow_test
+@testing.requires_testing_data
+def test_rap_music():
+    """Test RAP-MUSIC with evoked data
+    """
+    evoked, noise_cov, forward, forward_surf_ori, forward_fixed =\
+        _get_data()
+
+    def _check_dipoles(dipoles):
+        assert_true(len(dipoles), n_dipoles)
+        n_times = len(dipoles[0].times)
+        assert_true(dipoles[0].pos.shape[0], n_times)
+        assert_true(dipoles[0].ori.shape[0], n_times)
+
+        assert_true(dipoles[0].pos.shape[1], 3)
+        assert_true(dipoles[0].ori.shape[1], 3)
+
+    n_dipoles = 2
+
+    dipoles = rap_music(evoked, forward, noise_cov, n_dipoles=n_dipoles)
+    _check_dipoles(dipoles)
+
+    # Test with fixed forward
+    dipoles_fixed, res = rap_music(evoked, forward_surf_ori, noise_cov,
+                                   n_dipoles=n_dipoles,
+                                   return_residual=True)
+    _check_dipoles(dipoles_fixed)
+
+    # Test the residual times
+    assert_array_almost_equal(evoked.times, res.times)
+
+
+@slow_test
+@testing.requires_testing_data
+def test_rap_music_simulated():
+    """Test RAP-MUSIC with simulated evoked
+    """
+    evoked, noise_cov, forward, forward_surf_ori, forward_fixed =\
+        _get_data()
+
+    n_dipoles = 2
+    sim_evoked, stc = simu_data(evoked, forward_fixed, noise_cov, n_dipoles,
+                                evoked.times)
+
+    def _check_dipoles(dipoles, ori=False):
+        src = forward_fixed['src']
+        pos1 = forward_fixed['source_rr'][np.where(src[0]['vertno'] ==
+                                          stc.vertices[0])]
+        pos2 = forward_fixed['source_rr'][np.where(src[1]['vertno'] ==
+                                          stc.vertices[1])[0] +
+                                          len(src[0]['vertno'])]
+
+        # Check the position of the two dipoles
+        assert_true(dipoles[0].pos[0] in np.array([pos1, pos2]))
+        assert_true(dipoles[1].pos[0] in np.array([pos1, pos2]))
+
+        ori1 = forward_fixed['source_nn'][np.where(src[0]['vertno'] ==
+                                          stc.vertices[0])]
+        ori2 = forward_fixed['source_nn'][np.where(src[1]['vertno'] ==
+                                          stc.vertices[1])[0] +
+                                          len(src[0]['vertno'])]
+
+        if ori:
+            # Check the orientation of the two dipoles
+            assert_true(dipoles[0].ori[0] in np.array([ori1, ori2]))
+            assert_true(dipoles[1].ori[0] in np.array([ori1, ori2]))
+
+    # Check dipoles for fixed ori
+    dipoles = rap_music(sim_evoked, forward_fixed, noise_cov,
+                        n_dipoles=n_dipoles)
+    _check_dipoles(dipoles, ori=True)
+
+    # Check dipoles for free ori
+    dipoles = rap_music(sim_evoked, forward_fixed, noise_cov,
+                        n_dipoles=n_dipoles)
```

Comment:
you should use an approximate ori test for free ori case.

also be careful you use forward_fixed here too :(